pkg/testutils/watcher.go

```diff
@@ -104,3 +151,30 @@ func GetCACertificate(ctx context.Context, namespacedCertName types.NamespacedNa
 
 	return caCert, nil
 }
+
+func GetTLSSecret(ctx context.Context, namespacedSecretName types.NamespacedName, k8sClient client.Client,
+) (*apicorev1.Secret, error) {
+	tlsSecret := &apicorev1.Secret{}
+	if err := k8sClient.Get(ctx, namespacedSecretName, tlsSecret); err != nil {
+		return nil, fmt.Errorf("failed to get secret %w", err)
+	}
+
+	return tlsSecret, nil
+}
+
+func GatewaySecretCreationTimeIsUpdated(ctx context.Context, oldTime time.Time, kcpClient client.Client) error {
+	gwSecret, err := gatewaysecret.GetGatewaySecret(ctx, kcpClient)
+	if err != nil {
+		return fmt.Errorf("failed to get gateway secret %w", err)
+	}
+
+	currentTime, err := gatewaysecret.GetValidLastModifiedAt(gwSecret)
+	if err != nil {
+		return fmt.Errorf("failed to get last modified time %w", err)
+	}
+
+	if currentTime != oldTime {
```

Comment:
I think to only compare different is not solid enough, currentTime should larger than oldTime.